api/v1alpha1/tenantcontrolplane_types.go

```diff
@@ -90,6 +90,20 @@ type ServiceSpec struct {
 	ServiceType ServiceType `json:"serviceType"`
 }
 
+// AddonSpec defines the spec for every addon.
+type AddonSpec struct {
+	// +kubebuilder:default=true
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+// AddonsSpec defines the enabled addons and their features.
+type AddonsSpec struct {
+	// +kubebuilder:default={enabled: true}
+	CoreDNS AddonSpec `json:"coreDNS,omitempty"`
+	// +kubebuilder:default={enabled: true}
+	KubeProxy AddonSpec `json:"kubeProxy,omitempty"`
```

Comment:
Are these defaults required although specified in `TenantControlPlaneSpec.Addons` struct key?